relay-general/src/protocol/security_report.rs

```diff
@@ -216,22 +216,34 @@ impl CspRaw {
         "self".to_string()
     }
 
-    fn effective_directive(&self) -> Result<CspDirective, InvalidSecurityError> {
+    fn effective_directive(&self) -> Result<(CspDirective, String), InvalidSecurityError> {
         // Firefox doesn't send effective-directive, so parse it from
-        // violated-directive but prefer effective-directive when present
-        //
+        // violated-directive but prefer effective-directive when present.
         // refs: https://bugzil.la/1192684#c8
 
-        if let Some(directive) = self.effective_directive {
-            return Ok(directive);
+        if let Some(directive) = &self.effective_directive {
+            // In C2P1 and CSP2, violated_directive and possibly effective_directive might contain
+            // more information than just the CSP-directive. For that reason, we try to split it
+            // and we return the original value as well so as to not lose any information.
+            if let Ok(parsed_directive) = directive
+                .split_once(' ')
```

Comment:
so 'effective_directive' isn't supposed to have the old longer format even in csp1 and csp2, although we have a customer who populates this field with that format anyway, so maybe we should also apply the same logic to it.